dprint.py

```diff
@@ -50,18 +52,18 @@ def get_plugin_infos(self, dir_path):
 				raise Exception("Your installed version of dprint is out of date. Please update it.")
 
 		return editor_info["plugins"]
+
 	def format_text(self, dir_path, file_path, file_text):
 		from subprocess import Popen, PIPE
-		p = Popen(["dprint", "stdin-fmt", "--file-name", os.path.basename(file_path)],
-			stdin=PIPE, stdout=PIPE, stderr=PIPE, cwd=dir_path, shell=True)
+
+		p = Popen(["dprint", "fmt", "--stdin", os.path.basename(file_path)],
+			stdin=PIPE, stdout=PIPE, stderr=PIPE, cwd=dir_path)
 		stdout, stderr = p.communicate(bytes(file_text, "utf8"))
+
 		if p.returncode == 0:
-			result = stdout.decode("utf8")
-			if result.endswith("\r\n"):
-				return result[0:-2]
-			elif result.endswith("\n"):
-				return result[0: -1]
-			else:
-				return result
+			return stdout.decode("utf8")
 		else:
 			raise Exception("Error formatting: " + stderr.decode("utf8"))
+
+dprint_exec = DprintExec()

```

Comment:
Thanks! This is me not knowing python :)

Reply:
I'm not great at it either. Took a bit of trial and error and digging in dprint's code to get this working.